### ISSUE TITLE
making HDR optional instead of forced

### ISSUE THREAD

**Issue by Gammler** [DuoStream/Duo#31]
Hello,

i have a question about a feature and how the HDR function is implemented which is not really related to the other Issue i opened.

As you might now, displaying content as pure as possible without any conversion before it is presented on the screen is the best way to use the source material with your TV or display.

- For example frame rate matching, forced or VRR eliminates the need for frame rate conversion, which in the end will almost always result in frame drops or duplicating frames making motion less fluid.
- Color conversion should always be avoided, the display in the best case should aways present the exact data from the original source. This eleminates in a lot of cases wrong conversions.
- any tinkering with the source also can increase latency
- i'm sure there are other examples.

In short just present the source as it came from the original data. (colorspace, framerate matching, audio matching etc.)

With DUOs latest HDR implementation i see a problem right now. Because i can only tell it to run in SDR or HDR.
But with Sunshine standalone i can tell it to let HDR kick in as soon as the source display switches HDR on.
This produces the the first obvious problem of being faced with the flaws of SDR to HDR conversion which windows is still doing a bad job to this day. You can notice this right away when SDR content is shown with HDR enabled.

Is there something that can be done about this? At this point i would be forced to use two profiles and always switch between them, which would be kinda annoying to be honest.

Best regards

**Comment by Black-Seraph** [comment]
This is a limitation of remote IddCx drivers.

They use a different API for mode-switching than console IddCx drivers.

It's late here, but I'll give you a full documentary write-up of the details sometime tomorrow.

It can be fixed to an extend but not in the way you describe.
